src/main/java/seedu/address/model/task/EndTimeComparator.java

```diff
@@ -0,0 +1,21 @@
+package seedu.address.model.task;
+
+import java.util.Comparator;
+
+public class EndTimeComparator implements Comparator<ReadOnlyTask> {
+    public int compare(ReadOnlyTask p, ReadOnlyTask q) {
+        if (p.getEndTime().isPresent() && q.getEndTime().isPresent()) {
+            if (p.getEndTime().get().dateTime.isBefore(q.getEndTime().get().dateTime)) {
+                return -1;
+            } else if (p.getEndTime().get().dateTime.isAfter(q.getEndTime().get().dateTime)) {
```

Comment:
These predicates should be extracted out to a variable to make it more understandable and readable.

e.g. `boolean hasStartAndEnd = p.getStartTime().isPresent() && p.getEndTime().isPresent();`